src/Mutator/Boolean/TrueValue.php

```diff
@@ -54,12 +60,29 @@ public function mutate(Node $node)
         return new Node\Expr\ConstFetch(new Node\Name('false'));
     }
 
-    protected function mutatesNode(Node $node): bool
+    protected function mutatesNode(Node $node, array $mutatorSettings): bool
```

Comment:
What if we move the 'settings check' to a separate method, like `isAllowedBySettings(Node $node): bool`.

Then the abstract Mutator class simly has a return true there, and if a mutator has settings, then it can override that method.

Reply:
I would keep the number of the methods in Mutator class as few as possible. Also, I think `isAllowedBySettings()` is very coupled with `mutatesNode()` and almost always will need to do the same things in the beginning (like checking if it's a `ConstFetch` in `TrueValue` mutator), so we will end up with code duplication